src/python.gen.ts

```diff
@@ -252,7 +265,7 @@ ${this.hooks.join('\n')}
    */
   construct(indent: string, type: IType) {
     // Skip read-only parameters
-    if (!(type instanceof WriteType)) return ''
+    if (!this.inputTypes.has(type)) return ''
     indent = this.bumper(indent)
     const bump = this.bumper(indent)
     let result = `\n\n${indent}def __init__(self, *, `
```

Comment:
Maybe we should have this line wrap and indent so the first "real" parameter is indented with the rest of the parameters?

Reply:
yeah, that would make the tests look prettier. I think I tried once and failed but I can try again. Once `black` reformats the file it all looks good so we ship beautiful code :-)